File: src/main/java/org/example/plugin/SampleEventHandler.java
package org.example.plugin;

import dev.arbjerg.lavalink.api.ISocketContext;
import dev.arbjerg.lavalink.api.PluginEventHandler;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
public class SampleEventHandler implements PluginEventHandler {

    private static final Logger log = LoggerFactory.getLogger(SampleEventHandler.class);

    public SampleEventHandler() {
        log.info("Hello, world!");
    }

    @Override
    public void onWebSocketOpen(ISocketContext context, boolean resumed) {
        log.info("Websocket opened!");
    }
}

File: src/main/java/org/example/plugin/SampleEventHandler.kt
package org.example.plugin

import dev.arbjerg.lavalink.api.ISocketContext
import dev.arbjerg.lavalink.api.PluginEventHandler
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class SampleEventHandler : PluginEventHandler() {
    init {
        log.info("Hello, world!")
    }

    override fun onWebSocketOpen(context: ISocketContext, resumed: Boolean) {
        log.info("Websocket opened!")
    }

    companion object {
        private val log = LoggerFactory.getLogger(SampleEventHandler::class.java)
    }
}